python/tests/test_highlevel.py

```diff
@@ -1660,6 +1661,97 @@ def test_newick(self):
             for tree in ts.trees():
                 self.verify_newick(tree)
 
+    def test_as_dict_of_dicts(self):
+        for ts in get_example_tree_sequences():
+            tree = next(ts.trees())
+            print(tree)
+
+            adj_dod = tree.as_dict_of_dicts()
+            print(adj_dod)
+
+            g = nx.DiGraph(adj_dod)
+            print(g.nodes)
+
+            self.verify_nx_graph_topology(tree, g)
+            self.verify_nx_algorithm_equivalence(tree, g)
+            self.verify_nx_for_tutorial_algorithms(tree, g)
+
+    def verify_nx_graph_topology(self, tree, g):
+        self.assertSetEqual(set(tree.nodes()), set(g.nodes))
+
+        self.assertSetEqual(
+            set(tree.roots),
+            {n for n in g.nodes if g.in_degree(n) == 0}
+        )
+
+        self.assertSetEqual(
+            set(tree.leaves()),
+            {n for n in g.nodes if g.out_degree(n) == 0}
+        )
+
+        # test if tree has no in-degrees > 1
+        self.assertTrue(nx.is_branching(g))
+
+    def verify_nx_algorithm_equivalence(self, tree, g):
+        for root in tree.roots:
+            print(root)
+            self.assertTrue(nx.is_directed_acyclic_graph(g))
+
+            # test descendants
+            self.assertSetEqual(
+                set(u for u in tree.nodes() if tree.is_descendant(u, root)),
+                set(nx.descendants(g, root)) | {root}
+            )
```

Comment:
Is it a bug that root nodes are considered descendants of themselves by `Tree.is_descendant`?